include/pybind11/detail/smart_holder_poc.h

```diff
@@ -99,13 +110,16 @@ guarded_delete make_guarded_builtin_delete(bool armed_flag) {
 }
 
 template <typename T, typename D>
-inline void custom_delete(void *raw_ptr) {
-    D()(static_cast<T *>(raw_ptr));
-}
+struct custom_deleter {
+    D deleter;
+    explicit custom_deleter(D &&deleter) : deleter{std::move(deleter)} {}
```

Comment:
This is always tricky, but shouldn't this `std::move` be a `std::forward` instead? THe `D&&` could be a universal reference to an lvalue I think, in which case moving it may not have the intended effect at the call site, while forward will move if it is an rvalue but copy if it is an lvalue?

Reply:
Changed, thanks for catching this!

> This is always tricky,

Yeah, I wasn't paying enough attention here before.